src/app/content/modal/modal/modal.component.ts

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@angular/core';
+import {Observable, take} from 'rxjs';
+import {Store} from "@ngrx/store";
+import {AppState} from "../../../header/state/user/user.state";
+import {selectIsCreated, selectToggle, selectUserName} from "../../../header/state/user/user.selectors";
+import * as UserActions from "../../../header/state/user/user.actions"
+import {Router} from "@angular/router";
+
+
+@Component({
+  selector: 'app-modal',
+  templateUrl: './modal.component.html',
+})
+export class ModalComponent {
+  showModal$: Observable<boolean> = new Observable<boolean>();
+  userName$: Observable<string> = new Observable<string>();
+  isCreated$: Observable<boolean> = new Observable<boolean>();
+
+  isShown = false;
+  isCreated = true;
+
+  constructor(private store: Store<AppState>, private router: Router) {
+    this.showModal$ = this.store.select(selectToggle);
+    this.userName$ = this.store.select(selectUserName);
+    this.isCreated$ = this.store.select(selectIsCreated);
+
+    this.showModal$.subscribe(isShown => {
+      this.isShown = isShown;
+    })
+  }
+
+  toggleModal() {
+    this.store.dispatch(UserActions.showModal({showModal: this.isShown}));
+  }
+
+  userIsCreated() {
+    this.isCreated$.pipe(take(1)).subscribe(isCreated => {
+      this.isCreated = isCreated;
+      this.store.dispatch(UserActions.isCreatedUser({isCreated: isCreated}));
+      if (isCreated) {
```

Comment:
it's not guaranteed that isCreaeted is already evaluated properly in between line 39 and line 40 !

you should [write an effect for this](https://ngrx.io/guide/effects#functional-effects)

1) listen to the action with "ofType"
2) use concatLatestFrom to use a selector inside the effect

```ts
export const displayErrorAlert = createEffect(
  (actions$ = inject(Actions), store = inject(Store)) => {
    return actions$.pipe(
      ofType(UserActions.isCreatedUser),
      concatLatestFrom(() =>store.select(selectIsCreated)
      tap(({ ... }) => ...)
    );
  },
  { functional: true, dispatch: false }
);

```

Reply:
i decided to solve it in reducer just by adding:    on(UserActions.isCreatedUser, (state, { isCreated }) => ({
    ...state,
    isCreated: isCreated,
    **showModal: isCreated ? !state.showModal : state.showModal**
  })),